core/templates/services/prevent-reload-event.service.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for giving warning before leaving page.
+ */
+
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { WindowRef } from 'services/contextual/window-ref.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PreventReloadEvent {
+  listenerActive: boolean;
```

Comment:
Do we need this to be public?

Reply:
It can be used to check if a listener is already active. Might be used by someone in the future.

Reply:
Will create a getter function instead